lib/index.d.ts

```diff
@@ -0,0 +1,264 @@
+  NumericLessThan?: number;
+  NumericLessThanPath?: string;
+  NumericLessThanEquals?: number;
+  NumericLessThanEqualsPath?: string;
+  StringEquals?: string;
+  StringEqualsPath?: string;
+  StringGreaterThan?: string;
+  StringGreaterThanPath?: string;
+  StringGreaterThanEquals?: string;
+  StringGreaterThanEqualsPath?: string;
+  StringLessThan?: string;
+  StringLessThanPath?: string;
+  StringLessThanEquals?: string;
+  StringLessThanEqualsPath?: string;
+  StringMatches?: string;
+  TimestampEquals?: string;
+  TimestampEqualsPath?: string;
+  TimestampGreaterThan?: string;
+  TimestampGreaterThanPath?: string;
+  TimestampGreaterThanEquals?: string;
+  TimestampGreaterThanEqualsPath?: string;
+  TimestampLessThan?: string;
+  TimestampLessThanPath?: string;
+  TimestampLessThanEquals?: string;
+  TimestampLessThanEqualsPath?: string;
+};
+
+type ChoiceRuleNot = {
+  Not: ChoiceRuleComparison;
+  Next: string;
+  Comment?: string;
+};
+
+type ChoiceRuleAnd = {
+  And: ChoiceRuleComparison[];
+  Next: string;
+  Comment?: string;
+};
+
+type ChoiceRuleOr = {
+  Or: ChoiceRuleComparison[];
+  Next: string;
+  Comment?: string;
+};
+
+type ChoiceRuleSimple = ChoiceRuleComparison & {
+  Next: string;
+  Comment?: string:
+};
+
+type ChoiceRule = ChoiceRuleSimple | ChoiceRuleNot | ChoiceRuleAnd | ChoiceRuleOr;
+
+interface Choice extends StateBase {
+  Type: 'Choice';
+  Choices: ChoiceRule[];
+  Default?: string;
+}
+
+interface Fail extends StateBase {
+  Type: 'Fail';
+  Cause?: string;
+  Error?: string;
+}
+
+interface Map extends StateBase {
+  Type: 'Map';
+  ItemsPath?: string;
+  Iterator: Definition;
+  ItemSelector?: {
+    [key: string]: string;
+  }
+  MaxConcurrency?: number;
+  Parameters?: {
+    [key: string]: string;
+  };
+}
+
+type Resource = string | { 'Fn::GetAtt': [string, 'Arn'] } | { 'Fn::Join': [string, Resource[]] };
+
+interface TaskParametersForLambda {
+  FunctionName?: Resource;
+  Payload?: {
+    'token.$': string;
+    [key: string]: string;
+  };
+  [key: string]: unknown;
+}
+
+interface TaskParametersForStepFunction {
+  StateMachineArn: Resource;
+  Input?: {
+    'AWS_STEP_FUNCTIONS_STARTED_BY_EXECUTION_ID.$': '$$.Execution.Id';
+    [key: string]: string;
+  };
+  Retry?: [{ ErrorEquals?: string[] }];
+  End?: boolean;
+}
+
+// https://docs.aws.amazon.com/step-functions/latest/dg/connect-sns.html
+interface TaskParametersForSNS {
+  TopicArn: Ref | string;
+  Message?: string;
+  Subject?: string;
+  // 'Message.$': '$.input.message';
+  MessageAttributes?: Record<string, any>;
+}
+
+interface Task extends StateBase {
+  Type: 'Task';
+  Resource: Resource;
+  Parameters?:
+  | TaskParametersForLambda
+  | TaskParametersForStepFunction
+  | TaskParametersForSNS
+  | { [key: string]: string | { [key: string]: string } };
+}
+
+interface Pass extends StateBase {
+  Type: 'Pass';
+  Parameters?: {
+    [key: string]: string | Array<unknown> | { [key: string]: string };
+  };
+  Result?: {
```

Comment:
```suggestion
  Result?: string | {
```
```suggestion
  Result?: string | [] | {
```
I'm using `Result: "${env:something}"` to add an array. Not sure what's the best approach here. The string value resolves into an array. The tip from the Ui console says: `Must be valid JSON.` Array and object are the only ones valid.

Reply:
Should we try something like this: https://github.com/microsoft/TypeScript/issues/1897#issuecomment-331765301

```ts
export type JSONPrimitive = string | number | boolean | null;
export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
export type JSONObject = { [member: string]: JSONValue };
export interface JSONArray extends Array<JSONValue> {}
```

```suggestion
  Result?: JSONValue;
```